### PR TITLE
Add a rst doc for the performance dashboard

### PR DESCRIPTION
**Pull request by desertfire**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100592

### CODE REVIEW

**Review by msaroufim — APPROVED** [review]
This error https://github.com/pytorch/pytorch/actions/runs/4877081812/jobs/8701522315?pr=100592#step:8:1424 means you need to add your doc here https://github.com/pytorch/pytorch/blob/main/docs/source/index.rst

And thank you this is super helpful

File: docs/source/compile/performance-dashboard.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+All the dashboard tests are defined in this
+`function <https://github.com/pytorch/pytorch/blob/3e18d3958be3dfcc36d3ef3c481f064f98ebeaf6/.ci/pytorch/test.sh#L305>`__.
+The exact test configurations are subject to change, but at the moment, we measure both inference and training
+performance with AMP precision on the three benchmark suites. We also measure different settings of TorchInductor,
+including ``default``, ``with_cudagraphs (default + cudagraphs)``, and ``dynamic (default + dynamic_shapes)``. 
+
+Can I check if my PR affects TorchInductor's performance on the dashboard before merging?
+-----------------------------------------------------------------------------------------
+
+Individual dashboard runs can be triggered manually by clicking the ``Run workflow`` button
```
Comment (huydhn) [inline docs/source/compile/performance-dashboard.rst:36]: Just a note FYI, first time contributor won't be able to use this, which is expected.

**Review by huydhn — APPROVED** [review]
Thank you for putting this up!

**Pull request closed without merging**